src/Generator/Resolver/Value/Chainable/FixtureReferenceResolver.php

```diff
@@ -147,12 +147,16 @@ private function resolveReferredFixture(
 
             $context->markIsResolvingFixture($referredFixtureId);
             $objects = $this->generator->generate($referredFixture, $fixtureSet, $context);
-            $fixtureSet =  $fixtureSet->withObjects($objects);
 
             if (false === $needsCompleteGeneration) {
+                $generatedObject = $objects->get($referredFixture);
+                $objects = $objects->with(new CompleteObject($generatedObject));
```

Comment:
I'm a bit confused by this. IIRC the `$needsCompleteGeneration` flag is just to say "get this object, but fully instantiated". So your added code reads as "if the object doesn't need to be complete, mark it as complete" and that wouldn't make sense

Reply:
hmm, for me it reads like "the object does not need to be completed furthermore, so first, mark it as completed and second unmark the needs completion flag ... because its already completed...

the other way around, if i would put that code elsewhere, it would read like "the object is still marked as "needs completion" but i would already mark it as completed, but why? 

i am not 100% sure about this flag... all i could find out from the source until now the flag seems to be only used in this very special constellation, to make it possible to hydrate the object although we are in the first pass... and if thats the point, then saving the "needsCompleteGeneration" flag and setting it back later would make no use at all... because what we want in this special situation is to have the object to be fully complete?!